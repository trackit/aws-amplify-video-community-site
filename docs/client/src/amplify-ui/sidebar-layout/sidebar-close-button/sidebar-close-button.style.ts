import {css} from "emotion";
import {MQTablet} from "../../styles/media";

export const sidebarCloseButtonStyle = css`
  width: 2rem;
  margin-right: 0.75rem;
  height: 100%;
  background-color: var(--amplify-background-color);

  &:hover {
    cursor: pointer;
  }

  display: none;
  ${MQTablet} {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  > a > img {
    width: 1.5rem;
    height: 1.5rem;
    float: right;
  }
`;
