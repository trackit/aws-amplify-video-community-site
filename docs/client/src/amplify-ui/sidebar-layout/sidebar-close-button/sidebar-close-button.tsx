import {Component, Prop, h, Host, State} from "@stencil/core";
import {sidebarLayoutContext} from "../sidebar-layout.context";
import {ToggleInView} from "../sidebar-layout.types";
import {sidebarCloseButtonStyle} from "./sidebar-close-button.style";
import {DESKTOP_CLOSE} from "../../../constants/img";

@Component({tag: "amplify-sidebar-close-button"})
export class AmplifySidebarCloseButton {
  /*** toggles the state provided by `sidebar-layout` */
  @Prop() readonly toggleInView?: ToggleInView;
  /*** whether or not the sidebar is in view, provided by `sidebar-layout` */
  @Prop() readonly inView?: boolean;
  /*** whether or not the button is being hovered */
  @State() isHovered = false;

  onClick = (e: Event) => {
    if (this.inView) {
      e.stopPropagation();
    }
    if (this.toggleInView) {
      this.toggleInView();
    }
  };

  hover = () => {
    this.isHovered = true;
  };

  unhover = () => {
    this.isHovered = false;
  };

  render() {
    const imgLink = this.isHovered
      ? DESKTOP_CLOSE.darkSrc
      : DESKTOP_CLOSE.lightSrc;
    return (
      <Host>
        <button
          class={sidebarCloseButtonStyle}
          onClick={this.onClick}
          onMouseEnter={this.hover}
          onMouseLeave={this.unhover}
        >
          <img alt={DESKTOP_CLOSE.alt} src={imgLink} />
        </button>
      </Host>
    );
  }
}

sidebarLayoutContext.injectProps(AmplifySidebarCloseButton, [
  "inView",
  "toggleInView",
]);
