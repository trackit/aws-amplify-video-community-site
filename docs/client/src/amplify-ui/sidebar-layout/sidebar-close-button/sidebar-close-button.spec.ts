import {AmplifySidebarCloseButton} from "./sidebar-close-button";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout-toggle", () => {
  describe("Component logic", () => {
    let amplifySidebarCloseButton: AmplifySidebarCloseButton;
    beforeEach(
      () => (amplifySidebarCloseButton = new AmplifySidebarCloseButton()),
    );

    it("should init `toggleInView` as `undefined`", () =>
      expect(amplifySidebarCloseButton.toggleInView).toBeUndefined());

    it("should init `inView` as `undefined`", () =>
      expect(amplifySidebarCloseButton.inView).toBeUndefined());

    it("should init `isHovered` as `false`", () =>
      expect(amplifySidebarCloseButton.isHovered).toBe(false));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarCloseButton],
            html: `<amplify-sidebar-close-button />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
