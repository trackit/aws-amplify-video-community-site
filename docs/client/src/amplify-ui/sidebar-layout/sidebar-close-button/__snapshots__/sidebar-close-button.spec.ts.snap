// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout-toggle Render logic should render 1`] = `
<amplify-sidebar-close-button>
  <button class="css-1uzsahx">
    <img alt="Close menu" src="/assets/close-light.svg">
  </button>
</amplify-sidebar-close-button>
`;
